src/Host/Client/Impl/rtvs/R/util.R

```diff
@@ -167,4 +167,24 @@ export_to_pdf <- function(width, height) {
     dev.copy(device=pdf,file=filepath,width=width,height=height)
     dev.off()
     readBin(filepath, 'raw', file.info(filepath)$size)
-}
\ No newline at end of file
+}
+
+# Helper to publish rmarkdown files remotely
+rmarkdown_publish <- function(blob_id, output_filename, output_format, encoding) {
+    # Create temp file to store the rmarkdown file
+    rmdpath <- tempfile('rmd_', fileext = '.rmd');
+    on.exit(unlink(rmdpath));
+    writeBin(get_blob(blob_id), rmdpath);
+
+    # Create temp directory to collect the generated files
+    output_filepath <- paste0(tempdir(), '/', output_filename);
+
+    rmarkdown::render(rmdpath, output_format = output_format, output_file = output_filepath,  output_dir = tempdir(), encoding = encoding);
+
+	output_id <- create_blob(readBin(output_filepath, 'raw', file.info(output_filepath)$size));
+
+	output_files <- list();
```

Comment:
You can do this all in one go like so:
```r
... <- list(blob_ids = ..., file_names = ...)
```
Also, underscores rather than dots for identifiers.